src/tasks/worker.ts

```diff
@@ -244,11 +245,17 @@ export class Worker {
 				if (!executed) {
 					await setTimeout(this.interval);
 				}
-				this.poll();
+				if (this.running) {
+					this.poll();
+				}
 			}
 		})();
 	}
 
+	public stop(): void {
```

Comment:
I also would like your opinion @joshbwlng on this, do you think we should maybe make this async already just to future proof we needing to do some async work to stop tasks?

Reply:
I like the idea. I can see the argument against it, but I find it quite likely that we would want to change this to async in the future, so 👍 from me.